### PR TITLE
Clear Containers for Docker Engine (v1.8.3)

### PR DESCRIPTION
**Pull request by jamesodhunt**
This is a follow-on to the original Clear Containers for Docker PR #16163 .

The branch still uses the "clr" execution driver since docker has not yet migrated to calling out to runc (OCI).

### CODE REVIEW

**Comment by runcom** [comment]
@jamesodhunt is this PR for "educational & testing purpose" as the old one? (cit. https://github.com/docker/docker/pull/16163#issuecomment-139024189)

File: pkg/parsers/operatingsystem/operatingsystem_linux.go [hunk 1]
```diff
@@ -12,12 +13,22 @@ var (
 
 	// file to check to determine Operating System
 	etcOsRelease = "/etc/os-release"
+
+	// used by stateless systems like Clear Linux
+	altEtcOSRelease = "/usr/lib/os-release"
```
Comment (runcom) [inline pkg/parsers/operatingsystem/operatingsystem_linux.go]: see #17174 hope to get this merged to fix this :)

**Comment by runcom** [comment]
is this also supposed to work on non-clear-linux host? cause I get:
```
Error response from daemon: Cannot start container 287c38642d38f1b6bc00bd1f319068d234ff325454b75b7ae14e13a697243e46: fork/exec lkvm: no such file or directory
```
(following the ML I came across this last month https://lists.clearlinux.org/pipermail/dev/2015-September/000042.html)

**Comment by jamesodhunt** [comment]
Hi @runcom - since docker is not yet calling out to `runc`, we would like you to consider merging this branch as this is the only way to use an alternative runtime at present.

Note that we also have a basic OCI-compatible runtime for Clear Containers at:

- https://github.com/clearlinux/kvmtool/tree/oci.

Once docker starts calling an OCI runtime directly, we would be able to switch between the default and our runtime, using something like:

```
docker daemon --exec-opt "oci.runtime.cmd=lkvm" ...
```

At that point, this exec-driver could be dropped. However, it is unclear when docker will switch to OCI (docker 2.0?)

**Comment by runcom** [comment]
@jamesodhunt thanks for the info! Do you actually ship https://github.com/clearlinux/kvmtool/tree/oci with Clear Linux (`lkvm`) or are you using the original kvmtool from kernel.org?

**Comment by jamesodhunt** [comment]
@runcom - we are currently shipping the Clear Containers exec driver, and are awaiting an OCI-capable docker to validate our OCI-lkvm work before shipping it as stable.

**Comment by runcom** [comment]
I'm getting `DEBU[0504] Remove interface failed: Link not found` in daemon logs when doing a simple `docker run busybox echo "hello world" (Fedora 22)

**Comment by runcom** [comment]
`docker run busybox /bin/sh` gives you a shell prompt but the hangs also

**Pull request closed without merging**